src/Nethermind/Nethermind.Evm.Test/PointEvaluationPrecompileTests.cs

```diff
@@ -0,0 +1,95 @@
+// SPDX-FileCopyrightText: 2022 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System;
+using System.Collections;
+using FluentAssertions;
+using Nethermind.Core.Extensions;
+using Nethermind.Evm.Precompiles;
+using Nethermind.Specs.Forks;
+using NUnit.Framework;
+
+namespace Nethermind.Evm.Test;
+
+[TestFixture]
+public class PointEvaluationPrecompileTests
+{
+    [TestCaseSource(typeof(TestCases))]
+    public void Test_PointEvaluationPrecompile(string testName, byte[] input, bool expectedResult, long expectedGasSpent, byte[] expectedOutput)
+    {
+        IPrecompile precompile = PointEvaluationPrecompile.Instance;
+        (ReadOnlyMemory<byte> output, bool success) = precompile.Run(input, ShardingFork.Instance);
+        long gasSpent = precompile.DataGasCost(input, ShardingFork.Instance) + precompile.BaseGasCost(ShardingFork.Instance);
+
+        output.ToArray().Should().BeEquivalentTo(expectedOutput, testName);
+        success.Should().Be(expectedResult, testName);
+        gasSpent.Should().Be(expectedGasSpent, testName);
+    }
+
+    private static readonly byte[] _predefinedSuccessAnswer = Bytes.FromHexString("001000000000000001000000fffffffffe5bfeff02a4bd5305d8a10908d83933487d9d2953a7ed73");
+    private static readonly byte[] _predefinedFailureAnswer = new byte[0];
+    private static readonly long _fixedGasCost = 50000;
+
+    public class TestCases : IEnumerable
+    {
+        public IEnumerator GetEnumerator()
+        {
+            yield return new object[] {
+                "Incorrect data size",
+                Bytes.FromHexString("00"),
+                false,
+                _fixedGasCost,
+                _predefinedFailureAnswer
+            };
```

Comment:
Would be a little simpler to use IEnumerable property for source. And if enumerable would be returning TestCaseData, with proper name and maybe Result.